src/GShark/Geometry/Mesh.cs

```diff
@@ -548,6 +548,144 @@ private FaceData CountFaceEdges()
+                Line segment_CA = new Line(trianglePoints[2], trianglePoints[0]);
+                List<Line> Segments_list = new List<Line>() { segment_AB, segment_BC, segment_CA };
+
+                //Create a segment for each vertex
+                foreach (Line segment in Segments_list)
+                {
+                    Point3 SegmentClosestPoint = segment.ClosestPoint(point);
+                    EdgesClosestPoints.Add(SegmentClosestPoint);
+                }
+                Point3 edgeClosestPoint = CloudClosestPoint(point, EdgesClosestPoints);
+                ClosestPoint = edgeClosestPoint;
+            }
+
+            return ClosestPoint;
+        }
+
+        public Point3 MeshClosestPointTest(Point3 point)
+        {
+            //Gets vertices and converts them from type "MeshVertex" to "Point3"
+            List<MeshVertex> meshVertices = Vertices;
+            List<Point3> verticesPoints = meshVertices.ConvertAll(v => (Point3)v);
+
+            //Finds closer vertex to the target point
+            Point3 CloserPointVertex = CloudClosestPoint(point, verticesPoints);
+            //Finds the equivalent point as a type "MeshVertex" using the index of the Point3 in its equivalent list
+            MeshVertex CloserVertex = meshVertices[verticesPoints.IndexOf(CloserPointVertex)];
+
+            //Finds adjiacent faces to the closer vertex
+            System.Collections.Generic.IEnumerable<MeshFace> AdjacentFaces = CloserVertex.AdjacentFaces();
+
+            List<Point3> CloserPointToFace_List = new List<Point3>();
+
+            foreach (MeshFace face in AdjacentFaces)
+            {
+                //Retrives the Vertices associated with each face
+                List<MeshVertex> GSVertices = face.AdjacentVertices();
+                //Converts from MeshVertex to Point3
+                List<Point3> QuadPoints = GSVertices.ConvertAll(v => (Point3)v);
+
+                var trianglePoints1 = new List<Point3>() { QuadPoints[0], QuadPoints[1], QuadPoints[2] };
+                var trianglePoints2 = new List<Point3>() { QuadPoints[1], QuadPoints[2], QuadPoints[3] };
+                var trianglePoints3 = new List<Point3>() { QuadPoints[0], QuadPoints[1], QuadPoints[3] };
+                var trianglePoints4 = new List<Point3>() { QuadPoints[0], QuadPoints[2], QuadPoints[3] };
+
+                Point3 ClosestPoint1 = ClosestPointToTriangle(trianglePoints1, point);
+                Point3 ClosestPoint2 = ClosestPointToTriangle(trianglePoints2, point);
+                Point3 ClosestPoint3 = ClosestPointToTriangle(trianglePoints3, point);
+                Point3 ClosestPoint4 = ClosestPointToTriangle(trianglePoints4, point);
+
+                List<Point3> TrianglesCP = new List<Point3>() { ClosestPoint1, ClosestPoint2, ClosestPoint3, ClosestPoint4 };
+                Point3 ClosestPoint = CloudClosestPoint(point, TrianglesCP);
+
+                CloserPointToFace_List.Add(ClosestPoint);
+            }
```

Comment:
It seams that here you are assuming that the face is a quad face here?
What we want to do is to dispatch if a face is a triangle (then we use ClosestPointToTriangle and continue the foreach iteration), if it is not a triangle, we triangulate the face and we get the closest of the closestpoints of all subtriangles.